var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -848,14 +858,28 @@ def build(self, spec, prefix):
 
         bazel(*args)
 
-        build_pip_package = Executable("bazel-bin/tensorflow/tools/pip_package/build_pip_package")
-        buildpath = join_path(self.stage.source_path, "spack-build")
-        build_pip_package("--src", buildpath)
+        if self.spec.satisfies("@:2.16"):
+            build_pip_package = Executable(
+                "bazel-bin/tensorflow/tools/pip_package/build_pip_package"
+            )
+            buildpath = join_path(self.stage.source_path, "spack-build")
+            build_pip_package("--src", buildpath)
 
     def install(self, spec, prefix):
         tmp_path = env["TEST_TMPDIR"]
-        buildpath = join_path(self.stage.source_path, "spack-build")
-        with working_dir(buildpath):
-            args = std_pip_args + ["--prefix=" + prefix, "."]
-            pip(*args)
+        if self.spec.satisfies("@2.17:"):
+            buildpath = join_path(
+                self.stage.source_path,
+                "bazel-out/k8-opt/bin/tensorflow/tools/pip_package/wheel_house/",
```

Comment:
Will it always be k8-opt? Where does this come from? The [source build docs](https://www.tensorflow.org/install/source) don't mention it, nor does the package recipe.

Reply:
I'll look

Reply:
nope, won't always be that, will fix.